src/org/usfirst/frc/team2399/robot/subsystems/DriveTrain.java

```diff
@@ -4,35 +4,218 @@
 import org.usfirst.frc.team2399.robot.commands.JoyDrive;
 
 import com.ctre.CANTalon;
+import com.ctre.CANTalon.FeedbackDevice;
+import com.ctre.CANTalon.TalonControlMode;
 
+import edu.wpi.first.wpilibj.Talon;
+import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj.command.Subsystem;
 
 public class DriveTrain extends Subsystem {
+	
 	private CANTalon leftFrontTalon;
 	private CANTalon rightFrontTalon;
 	private CANTalon leftBackTalon;
 	private CANTalon rightBackTalon;
-
+	private CANTalon leftMiddleTalon;
+	private CANTalon rightMiddleTalon;
+	
+	private double goalDistance;
+	
+	private double anglePConstant = RobotMap.ANGLE_P_CONSTANT;
+	private double desiredAngle;
+	
 	public DriveTrain() {
 		leftFrontTalon = new CANTalon(RobotMap.DRIVETRAIN_LEFT_TALON_FRONT_ADDRESS);
 		rightFrontTalon = new CANTalon(RobotMap.DRIVETRAIN_RIGHT_TALON_FRONT_ADDRESS);
 		leftBackTalon = new CANTalon(RobotMap.DRIVETRAIN_LEFT_BACK_TALON_ADDRESS);
 		rightBackTalon = new CANTalon(RobotMap.DRIVETRAIN_RIGHT_BACK_TALON_ADDRESS);
+		leftMiddleTalon = new CANTalon(RobotMap.DRIVETRAIN_LEFT_MIDDLE_TALON_ADDRESS);
+		rightMiddleTalon = new CANTalon(RobotMap.DRIVETRAIN_RIGHT_MIDDLE_TALON_ADDRESS);
+		
+		/**
+		 * Sets the type of encoder to be routed through the Talon, as well as what mode the Talon is in
+		 */
+		leftFrontTalon.setFeedbackDevice(FeedbackDevice.CtreMagEncoder_Relative);
+		rightFrontTalon.setFeedbackDevice(FeedbackDevice.CtreMagEncoder_Relative);
+		
+		/**
+		 * Talons that don't have encoders are set to Follower mode
+		 * The line below that tells them which Talon to follow
+		 * They will go to whichever mode the Talon they're set to follow 
+		 */
+		leftMiddleTalon.changeControlMode(TalonControlMode.Follower);
+		leftMiddleTalon.set(RobotMap.DRIVETRAIN_LEFT_TALON_FRONT_ADDRESS);
+		leftBackTalon.changeControlMode(TalonControlMode.Follower);
+		leftBackTalon.set(RobotMap.DRIVETRAIN_LEFT_TALON_FRONT_ADDRESS);
+		rightMiddleTalon.changeControlMode(TalonControlMode.Follower);
+		rightMiddleTalon.set(RobotMap.DRIVETRAIN_RIGHT_TALON_FRONT_ADDRESS);
+		rightBackTalon.changeControlMode(TalonControlMode.Follower);
+		rightBackTalon.set(RobotMap.DRIVETRAIN_RIGHT_TALON_FRONT_ADDRESS);
+		
+		/**
+		 * If the forward constant is negative (see boolean in RobotMap) reverse the output of
+		 * either the sensor or the motor
+		 */
+		leftFrontTalon.reverseOutput(RobotMap.REVERSE_LEFT_FRONT_MOTOR_OUTPUT_CONSTANT);
+		leftFrontTalon.reverseSensor(RobotMap.REVERSE_LEFT_ENCODER_OUTPUT_CONSTANT);
+		rightFrontTalon.reverseOutput(RobotMap.REVERSE_RIGHT_FRONT_MOTOR_OUTPUT_CONSTANT);
+		rightFrontTalon.reverseSensor(RobotMap.REVERSE_RIGHT_ENCODER_OUTPUT_CONSTANT);
+		
+		/**
+		 * Sets all constants for PID loops
+		 * We won't be using I and D, and they are set to 0 automatically, but
+		 * for clarity and safety's sake, they're set to zero here
+		 * TODO: Set actual values for F and P
+		 */
+		leftFrontTalon.setF(0);
```

Comment:
This does not need to be completed for this review to pass, BUT I made a work item for experimentally determining these: https://trello.com/c/06T2jY5o